userspace/libscap/scap.c

```diff
@@ -269,7 +269,9 @@ scap_t* scap_open_live_int(char *error,
 	if((res = scap_proc_scan_proc_dir(handle, filename, -1, -1, NULL, error, true)) != SCAP_SUCCESS)
 	{
 		scap_close(handle);
-		snprintf(error, SCAP_LASTERR_SIZE, "error creating the process list. Make sure you have root credentials.");
+		char err[SCAP_LASTERR_SIZE] = {0};
```

Comment:
`SCAP_LASTERR_SIZE + 1` should be safer because strncpy may not null-terminate the string